### ISSUE TITLE
Query does not deliver any match

### ISSUE THREAD

**Issue by nlincke** [filiphanes/fts-elastic#10]
Hey,

I installed this plugin following the given guide. It works in a sense where it does not throw an error ;) 
But it also does not deliver any search result...
Inside the logs i can see the actual search query and after fiddling around with this query i could identify an issue:
```

....
"bool": {
      "filter": [
        {"term": {"user": "filip.hanes@example.com"}},
        {"term": {"box": "f40efa2f8f44ad54424000006e8130ae"}}
      ],
....
```

this part of the query always fails! Especially the `{"term": {"user": "filip.hanes@example.com"}}`, part.
If i delete this part of the query i can see matching (i think) results....

I can see that all my mails are inside my elastic search instance...
... has anyone any idea what i do wrong?

Cheers!

**Comment by filiphanes** [comment]
It may be that user doesn't match, I expect that you are actually trying your user not `filip.hanes@example.com`.
Otherwise you need to check:
 - elastic version
 - did you installed correct elastic mapping
 - did you run `doveadm fts rescan` after changing schema?
 - are you trying correct user
 - can you send whole log file (fts-elastic http rawlog in and out) or curl request which you are testing?
 - are you using routing key in you requests? are they correct?

**Comment by nlincke** [comment]
Hey,

thanks for your fast replay.
- Version is 7.8.0
- i hope so :D
- Sure, all data is there... If I "visit" http://localhost:9200/m/_search?pretty i see data... and if i query without the user field then i see matches...
- I guess so. The query is built by the plugin anyway. But the data is structured like this, right?:
```
        "_index": "m",
        "_type": "_doc",
        "_id": "...",
        "_score": 1.0,
        "_routing": "....",
        "_source": {
          "uid": 955,
          "box": "f40efa2f8f44ad54424000006e8130ae",
          "user": "filip.hanes@example.com",
          "date": "...",
          "from": "....",
          "to": "....",
          "message-id": "...",
          "subject": "...",
          "body": "...
```
Sure i used the mail and box from the example but my query fits to the data (i think) - i guess the user field in this plain data is the one targeted inside the query... right?
- yes, i am using the right user and yes i have a routing key in my request which matches the query user field and the user field of the data...

Example Request:
```
POST http://localhost:9200/m/_search?routing=filip.hanes@example.com
{
    "query":{
        "bool":{
            "filter":[{"term":{"user":"filip.hanes@example.com"}},{"term":{"box": "f40efa2f8f44ad54424000006e8130ae"}}],
            "must":[{"multi_match":{"query":"Party","operator":"and","fields":["from","to","cc","bcc","sender","subject","body"]}}]
            }
        }, 
    "size":10000, 
    "_source":false
}
```

result:
```
{
    "took": 23,
    "timed_out": false,
    "_shards": {
        "total": 1,
        "successful": 1,
        "skipped": 0,
        "failed": 0
    },
    "hits": {
        "total": {
            "value": 0,
            "relation": "eq"
        },
        "max_score": null,
        "hits": []
    }
}
```
This is also what i see inside the logs...
If i make the same query, using postman, like this:

```
POST http://localhost:9200/m/_search?routing=filip.hanes@example.com
{
    "query":{
        "bool":{
            "filter":[{"term":{"box": "f40efa2f8f44ad54424000006e8130ae"}}],
            "must":[{"multi_match":{"query":"Party","operator":"and","fields":["from","to","cc","bcc","sender","subject","body"]}}]
            }
        }, 
    "size":10000, 
    "_source":false
}
```
I get this:
```
{
    "took": 42,
    "timed_out": false,
    "_shards": {
        "total": 1,
        "successful": 1,
        "skipped": 0,
        "failed": 0
    },
    "hits": {
        "total": {
            "value": 43,
            "relation": "eq"
        },
        "max_score": 6.821225,
        "hits": [
            {
                "_index": "m",
                "_type": "_doc",
                "_id": "956/f40efa2f8f44ad54424000006e8130ae/filip.hanes@example.com",
                "_score": 6.821225,
                "_routing": "filip.hanes@example.com"
            },
            {
                "_index": "m",
                "_type": "_doc",
                "_id": "1058/f40efa2f8f44ad54424000006e8130ae/filip.hanes@example.com",
                "_score": 6.8030977,
                "_routing": "filip.hanes@example.com"
            },
....
```

**Comment by filiphanes** [comment]
ok, everything looks ok,
When you set `"_source":true` in query, does it return documents with `"user": "filip.hanes@example.com"`?
And what is your mapping `GET http://localhost:9200/m/_mapping` ?

**Comment by nlincke** [comment]
Hey,

If i run the query like this
```
POST http://localhost:9200/m/_search?routing=filip.hanes@example.com
{
    "query":{
        "bool":{
            "filter":[{"term":{"box": "f40efa2f8f44ad54424000006e8130ae"}}],
            "must":[{"multi_match":{"query":"Party","operator":"and","fields":["from","to","cc","bcc","sender","subject","body"]}}]
            }
        }, 
    "size":10000, 
    "_source":**true**
}
```
i see this:

```
{
    "took": 26,
    "timed_out": false,
    "_shards": {
        "total": 1,
        "successful": 1,
        "skipped": 0,
        "failed": 0
    },
    "hits": {
        "total": {
            "value": 43,
            "relation": "eq"
        },
        "max_score": 6.821225,
        "hits": [
            {
                "_index": "m",
                "_type": "_doc",
                "_id": "956/f40efa2f8f44ad54424000006e8130ae/filip.hanes@example.com",
                "_score": 6.821225,
                "_routing": "filip.hanes@example.come",
                "_source": {
                    "uid": 956,
                    "box": "f40efa2f8f44ad54424000006e8130ae",
                    "user": "filip.hanes@example.com",
                    "date": "...",
                    "from": "...",
                    "to": "...",
                    "message-id": "...",
                    "subject": "...",
                    "body": "..."
...
```
If i add `{"term":{"user":"filip.hanes@example.com"}},` to the query i get the same result as i would get with 
    `"_source":false`


here is my mapping
```
{
    "m": {
        "mappings": {
            "properties": {
                "bcc": {
                    "type": "text",
                    "fields": {
                        "keyword": {
                            "type": "keyword",
                            "ignore_above": 256
                        }
                    }
                },
                "body": {
                    "type": "text",
                    "fields": {
                        "keyword": {
                            "type": "keyword",
                            "ignore_above": 256
                        }
                    }
                },
                "box": {
                    "type": "text",
                    "fields": {
                        "keyword": {
                            "type": "keyword",
                            "ignore_above": 256
                        }
                    }
                },
                "cc": {
                    "type": "text",
                    "fields": {
                        "keyword": {
                            "type": "keyword",
                            "ignore_above": 256
                        }
                    }
                },
                "date": {
                    "type": "text",
                    "fields": {
                        "keyword": {
                            "type": "keyword",
                            "ignore_above": 256
                        }
                    }
                },
                "from": {
                    "type": "text",
                    "fields": {
                        "keyword": {
                            "type": "keyword",
                            "ignore_above": 256
                        }
                    }
                },
                "message-id": {
                    "type": "text",
                    "fields": {
                        "keyword": {
                            "type": "keyword",
                            "ignore_above": 256
                        }
                    }
                },
                "sender": {
                    "type": "text",
                    "fields": {
                        "keyword": {
                            "type": "keyword",
                            "ignore_above": 256
                        }
                    }
                },
                "subject": {
                    "type": "text",
                    "fields": {
                        "keyword": {
                            "type": "keyword",
                            "ignore_above": 256
                        }
                    }
                },
                "to": {
                    "type": "text",
                    "fields": {
                        "keyword": {
                            "type": "keyword",
                            "ignore_above": 256
                        }
                    }
                },
                "uid": {
                    "type": "long"
                },
                "user": {
                    "type": "text",
                    "fields": {
                        "keyword": {
                            "type": "keyword",
                            "ignore_above": 256
                        }
                    }
                }
            }
        }
    }
}
```

I would like to thank you already for your help :) I really appreciate it!

**Comment by filiphanes** [comment]
everything looks ok, what I know is that this is not fts-elastic fault, but something somewher some detail is wrong or you didn't noticed something
my last question: does it return results without routing key in url with `user` in query?

**Comment by nlincke** [comment]
No. It does not.... :/
if i delete the TLD and just search for the user part of the mail `filip.hanes` instead of `filip.hanes@example.com` address... it works... Does this information help some how?

**Comment by filiphanes** [comment]
hmm, try change mapping for user field to:
`"user": {
                            "type": "keyword",
                            "ignore_above": 256
                }`
because now it has text type.
or before you can try to search for `"user.keyword":"filip.hanes@example.com"`

**Comment by nlincke** [comment]
`user.keyword` works! :D
So changing the mapping should work? Shall i try it?

**Comment by filiphanes** [comment]
yes, try change mapping, I am not aware where fts-elastic would need to search text in user field
I would be thankfull if you would create a pull request

**Comment by nlincke** [comment]
Sure! Let me validate it and then I would love to do my part :) Thanks for helping

**Comment by nlincke** [comment]
Hey,

it is working now. And the description is working just fine! No change needed. 
Perhaps i forgot to attach the file as a payload or i did something else wrong but the index was created without any specific settings. I am not an elasticsearch expert, but i guess the settings i have found resulting form the actual usage of the plugin? Does this make sense? I was able to recreate the error by creating a new index without any specific settings...

Anyway, I recreated the index using the elastic7-schema.json in a proper way and now it is working... the mapping is correct :)

Thanks for the help and for the great plugin!
